Getting staging back in line with main

caparker opened the pull request.

jflasher commented on package.json:
```diff
@@ -25,6 +25,7 @@
     "babel-register": "^6.3.13",
     "boom": "^2.8.0",
     "csv-stringify": "0.0.8",
+    "date-fns": "^1.28.5",
```
We're already using moment.js throughout the project, can we use that here as well to avoid another library.

kamicut replied:
Ah, I didn't see `moment` in the package.json, but I could switch to it.

jflasher replied:
Yes, please! Would be nice to limit packages across the projects.

jflasher commented on api/controllers/latest.js:
```diff
@@ -67,7 +68,7 @@ if (process.env.USE_ATHENA) {
         value: Number(r.value),
         unit: r.unit,
-        date_utc: r.date.utc,
+        date_utc: hiveParse(r.date).utc,
```
I might be missing it, but where is `hiveParse()` coming from?

kamicut replied:
hiveParse is here https://github.com/openaq/openaq-api/blob/develop/lib/utils.js#L232

jflasher commented on api/controllers/countries.js:
```diff
@@ -19,7 +19,8 @@ var resultsQuery = db
 
 if (process.env.USE_ATHENA) {
-  resultsQuery = client.query('SELECT country, city, location, count(location) as count from fetches.fetches_realtime GROUP BY country, city, location ORDER BY country');
+  let query = 'SELECT country, city, location, count(location) as count from ' + client.fetchesTable + ' GROUP BY country, city, location ORDER BY country';
```
You can use JS template literals if you want, might read more nicely than this way.

kamicut replied:
Oddly enough I tried them but kept getting lint errors.

jflasher replied:
Huh, that's weird. Um, if you want to put them back in, I can fix up the linting errors.

kamicut replied:
I've added template strings, but couldn't get multiline template strings to work...

jflasher commented:
I added `ATHENA_FETCHES_TABLE=fetches.fetches_realtime_staging` to the staging env file.

jflasher commented:
Merging this in so we can run an aggregation on staging and see how it looks.

russbiggs committed:
Merge pull request #260 from openaq/staging

Release 20230913 (#259)

russbiggs committed:
Merge pull request #266 from openaq/staging

fix typo in auth variable (#265)

russbiggs committed:
Merge pull request #268 from openaq/staging

Hot fix/api key logging (#267)

russbiggs committed:
hot-fix resolve auth issues

russbiggs committed:
add redis in startup event hook

russbiggs committed:
Merge pull request #277 from openaq/staging

Release 20230920 (#274)

russbiggs committed:
Merge pull request #279 from openaq/staging

Staging

russbiggs committed:
Release 20230927 (#291)

* Added ability to increase the working memory for a given endpoint (#282) resolves #281

* Added ability to increase the working memory for a given endpoint  resolves #281

* update staging deploy to deploy on release branches (#284)

* Added timer (#280)

* included order_by in v1 and v2 measurements (#263)

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* manufacturers resource (#273)

* manufacturers resource resolves #252
---------

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* Adding `v3/instruments` resource (#271)

* instruments resource resolves #270 

---------

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* Added timer

* Added owner router

---------

Co-authored-by: Gabriel Fosse <67290377+majesticio@users.noreply.github.com>
Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>
Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* Add sort and order by query params to v3 endpoints  (#285)

* order by classes

* refactor query builder to handle sort and order by statements

* add sort and order by to v3 endpoints

resolves #163
---------

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* Feature/working memory query 281 (#287)



* Moved the tr.start out of if block

  Fixes bug #286

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* fix missing start variable and return empty string for order (#288)

* Remove count fields (#289)

* removed count fields
---------

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* fix failing unit test (#292)

* rate limiter logging fix (#293)

* remove summary count fields from v3 countries (#294)

* removed locations_count from instruments.py query

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>
Co-authored-by: Gabriel Fosse <67290377+majesticio@users.noreply.github.com>
Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

russbiggs committed:
revert to old startup and shutdown events (#296)

russbiggs committed:
add redis client to app state (#299)

russbiggs committed:
fixed parameters query param for locations, countries, and providers (#306) (#308)

* fixed parameters query param for locations, countries
and providers

* Made requested changes to implement
ParametersQuery

---------

Co-authored-by: Gabriel Fosse <67290377+majesticio@users.noreply.github.com>
Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

russbiggs committed:
add tiles parameter query (#310)

majesticio committed:
Update test_v3_queries.py file to use the array overlap operator instead of the equality operator. (#312)

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

russbiggs committed:
Fix/statement timeout (#316)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* set statement_timeout default config value

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

russbiggs committed:
Release 20231222 (#318)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* ratelimit headers (#315)

* ratelimit headers

* dynamic ratelimt-policy values

* Added wait_for method (#317)

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

russbiggs committed:
Release 20240104 (#320)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* ratelimit headers (#315)

* ratelimit headers

* dynamic ratelimt-policy values

* Fix/statement timeout (#316)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* set statement_timeout default config value

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* Added wait_for method (#317)

* update default date for date_form

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

caparker committed:
Merge remote-tracking branch 'upstream/main'

russbiggs committed:
Release 20240109 (#324)

* add hint to 408 timeout error (#322)

* Fix/bad timeout value (#323)

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* Bad timeout value passed to db class (#321)
  The averages endpoint was passing a config value but unassigned and
  it was therefor being passed as the timeout value. I added a check
  in the db to make sure we are protected from that as well as updated
  the averages endpoint to explicitly name the config argument.

fixes #321

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

russbiggs committed:
Readme redis update (#327)

* readme updates

* readme typo

russbiggs committed:
Release 20240207 (#331)

* Feature/ratelimiter update (#330)

* updated request_is_limited algorithm

* Strip out numbers from paths

  This is so we can better aggregate by path in cloudwatch

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

russbiggs committed:
update key setting for ratelimiting

russbiggs committed:
Release 20240210 (#334)

* Provide numeric value for dates for easier analysis later (#332)

* Feature/ratelimiter update (#330)

* updated request_is_limited algorithm

* Strip out numbers from paths

  This is so we can better aggregate by path in cloudwatch

* Provide numeric value for dates for easier analysis later

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* Sensors by location (#297)

* Added ability to increase the working memory for a given endpoint (#282) resolves #281

* Added ability to increase the working memory for a given endpoint  resolves #281

* update staging deploy to deploy on release branches (#284)

* Added timer (#280)

* included order_by in v1 and v2 measurements (#263)

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* manufacturers resource (#273)

* manufacturers resource resolves #252
---------

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* Adding `v3/instruments` resource (#271)

* instruments resource resolves #270 

---------

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* Added timer

* Added owner router

---------

Co-authored-by: Gabriel Fosse <67290377+majesticio@users.noreply.github.com>
Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>
Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* Add sort and order by query params to v3 endpoints  (#285)

* order by classes

* refactor query builder to handle sort and order by statements

* add sort and order by to v3 endpoints

resolves #163
---------

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* Feature/working memory query 281 (#287)



* Moved the tr.start out of if block

  Fixes bug #286

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* fix missing start variable and return empty string for order (#288)

* Remove count fields (#289)

* removed count fields
---------

Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* fix failing unit test (#292)

* rate limiter logging fix (#293)

* location sensors query

* location sensors path query

* Cleaned up the sensors endpoints

  Rewrote the queries to speed them up a litte and added the
  measurements method

* Removed old method that I had renamed

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>
Co-authored-by: Russ Biggs <russbiggs@gmail.com>
Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

* add deprecation warning (#333)

* upgrdate to actions/checkout@v4

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>
Co-authored-by: Gabriel Fosse <67290377+majesticio@users.noreply.github.com>
Co-authored-by: Gabriel Fosse <kraken+majesticio@users.noreply.github.com>

caparker committed:
Merge branch 'main' of github.com:openaq/openaq-api-v2

russbiggs committed:
Release 20240214 (#338)

* Feature/merge trends and measurements (#336)

* Fix/statement timeout (#316)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* set statement_timeout default config value

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* Release 20231222 (#318)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* ratelimit headers (#315)

* ratelimit headers

* dynamic ratelimt-policy values

* Added wait_for method (#317)

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* Added the trends methods (dow, hod, moy) to the measurements

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* Feature/add license to location (#337)

* Fix/statement timeout (#316)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* set statement_timeout default config value

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* Release 20231222 (#318)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* ratelimit headers (#315)

* ratelimit headers

* dynamic ratelimt-policy values

* Added wait_for method (#317)

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* Added the licsense list object to the locations

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

russbiggs committed:
Release 20240220 (#343)

* Feature/auth api (#340)

* auth api updates

* password reset (#341)

* Feature/password email (#342)

* password reset

* password email

* Feature/merge trends and measurements (#339)

* Fix/statement timeout (#316)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* set statement_timeout default config value

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* Release 20231222 (#318)

* Fix/v2 measurements source (#314)

* fix v2 measurements to query measurements table instead of hourly

* realias to match query params

* measurements geom fix

* Added sensors_id to speed up sorting

* update statement_timeout to milliseconds

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* ratelimit headers (#315)

* ratelimit headers

* dynamic ratelimt-policy values

* Added wait_for method (#317)

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

* Added the trends methods (dow, hod, moy) to the measurements

* Updated the trends and raw methods

  Added better period calculating for both

* Added validation check for dates

  Still has error related to serializing a datetime object

* Added work around for the value error + datetime issue

  Also fixed syntax errors in the bbox validation

---------

Co-authored-by: Russ Biggs <russbiggs@gmail.com>

* add placeholder for explorer_api_key

---------

Co-authored-by: Christian Parker <c.andrew.parker@gmail.com>

caparker committed:
Merge branch 'main' of github.com:openaq/openaq-api-v2

caparker committed:
Removed casting of datetime objects

caparker committed:
Added summary back

caparker committed:
Updated to use new `as_timestamptz` functions

russbiggs committed:
update prod ratelimiting

russbiggs committed:
v3 auth updates

caparker committed:
Restrict access to purpleair until we can remove data

caparker committed:
Changed tests to fit api

caparker committed:
Merge pull request #344 from openaq/feature/restrict-purpleair-access

Feature/restrict purpleair access

russbiggs committed:
add token register endpoint (#345) (#346)

caparker committed:
Allow the latest value to be empty (#351)

Fixes #349
